src/libstd/sys/hermit/stdio.rs

```diff
@@ -10,19 +10,21 @@ impl Stdin {
     pub fn new() -> io::Result<Stdin> {
         Ok(Stdin)
     }
+}
 
-    pub fn read(&self, data: &mut [u8]) -> io::Result<usize> {
+impl io::Read for Stdin {
+    fn read(&mut self, data: &mut [u8]) -> io::Result<usize> {
         self.read_vectored(&mut [IoSliceMut::new(data)])
     }
 
-    pub fn read_vectored(&self, _data: &mut [IoSliceMut<'_>]) -> io::Result<usize> {
+    fn read_vectored(&self, _data: &mut [IoSliceMut<'_>]) -> io::Result<usize> {
```

Comment:
This needs to be &mut self.

Reply:
Thanks for catching this. Do you know of any tests that actually attempt to compile libstd on these platforms? It seems like our CI doesn't catch them; this was only discovered by accident when there was an issue with a rollup commit: https://github.com/rust-lang/rust/pull/73475#issuecomment-646310446